service/s3/api_client.go

```diff
@@ -151,3 +173,7 @@ func addHTTPSignerV4Middleware(stack *middleware.Stack, o Options) {
 	signer := v4.Signer{}
 	stack.Finalize.Add(v4.NewSignHTTPRequestMiddleware(o.Credentials, signer), middleware.After)
 }
+
+func addUpdateEndpointMiddleware(stack *middleware.Stack, options Options) {
+	s3cust.UpdateEndpoint(stack, s3cust.UpdateEndpointOptions{UsePathStyle: options.UsePathStyle, UseAccelerate: options.UseAccelerate, Region: options.Region})
+}
```

Comment:
Region could be changed by some other middleware, probably want to not put region here and let the middleware access it from the context.